docs/dev/components/components.md

```diff
@@ -0,0 +1,94 @@
+# Components/Maven Modules
+
+Gaffer has lots of components, which can be split into different categories. This section of the developer doc provides more detail on the most important components.
+
+## Key components
+
+For more developer information on these key components, see their associated page.
+
+- [Operation](operation.md): Classes for the Operation interfaces and core operation implementations.
+- [Cache](cache.md): Classes for the Gaffer cache service.
+- [Graph](graph.md): Contains the Gaffer `Graph` object and related utilities.
+- [Data](data.md): Classes defining Gaffer's data objects: `Element`, `Edge` and `Entity`.
+- [Store](store.md): Contains the Gaffer `Store` object and related classes.
+- [Serialisation](serialisation.md): Contains classes for Serialisation in Gaffer. 
+- [Accumulo Store](accumulo-store.md): A store implemented using Apache Accumulo.
+- [Core REST](core-rest.md): Classes which provide a Gaffer REST API using Jersey/JAX-RS.
+- [Spring REST](spring-rest.md): Implementation of the Gaffer REST API deployed in a Spring Boot container.
+
+## Project Architecture (Maven)
```

Comment:
Maybe call this project structure instead of architecture? Architecture more implies the class interactions and core design of Gaffer where as, this section is actually just taking about the build framework and repo structure.

Reply:
Yes it might be good to include the Java architecture some time, though I think it should be refactored a bit first. I'll change this.